cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```diff
@@ -86,32 +168,81 @@ public void schedule(Id.Program program, SchedulableProgramType programType, Ite
 
   @Override
   public List<String> getScheduleIds(Id.Program program, SchedulableProgramType programType) {
-    return ImmutableList.of();
+    Set<String> scheduleIds;
+    synchronized (this) {
+      scheduleIds = ImmutableSet.copyOf(scheduleSubscribers.keySet());
+    }
+    ImmutableList.Builder<String> builder = ImmutableList.builder();
+    String programScheduleId = getProgramScheduleId(program, programType);
+    for (String scheduleId : scheduleIds) {
+      if (scheduleId.startsWith(programScheduleId)) {
+        builder.add(scheduleId);
```

Comment:
If it is an ordered concurrent map, then you could also do copyOf(submap.keySet())